docs/dev_guide/contents/ci_jobs.rst

```diff
@@ -0,0 +1,40 @@
+.. _ci_jobs:
+
+***************************************
+What runs on our Continuous Integration
+***************************************
+
+Overall the aim of the Continuous Integration is to provide a way for developers to test their code on multiple platforms and help in the development process.
+While we can run many jobs, we aim to limit the selection depending in the context to ensure that the time taken is not wasted or ends up delaying a pull request or release.
+
+Currently we have several stages of CI jobs, some run on a pull request and some will only run on a schedule.
+
+1. "core" - Pull Request, Scheduled and Release
+    This runs a basic offline test suite on Linux for the latest version of Python we support.
+    It ensures that basic functionality works and that we don't have any regressions.
+
+2. "test" - Pull Request, Scheduled and Release
+    This runs a basic test suite on Windows and Mac OS for older versions of Python we support.
+    It ensures that basic functionality works and that we don't have any regressions.
+    This stage needs to wait for the "core" stage to complete.
+
+3. "online" - Pull Request, Scheduled
+    This runs a full test suite on Linux for a version of Python.
+    This build can fail (due to a range of reasons) and can take up to 45 minutes to run.
+    We want to ensure that the offline tests are passing before we consider running this.
+    Therefore this stage needs to wait for the "core" stage to complete.
+    In addition, we run the documentation build that can fail due to the same reasons as the online build.
+
+    As these are not tested when we build the distribution or wheels, we skip these on a release.
+    These should be checked based on the last commit for that release branch instead before a tag.
+
+4. "cron" - Scheduled
+    Here we put builds that are useful to run on a schedule but not so useful on a pull request.
+    This allows us to run more "exotic" or focused builds that day to day, should not affect a pull request.
+
+    These are:
+
+    * "base_deps" - Check that sunpy does not error if you only have the base dependencies installed.
+    * "devdeps" - Check the offline test suite with the development dependencies installed.
+    * "oldestdeps" - Check the offline test suite with the oldest dependencies installed.
```

Comment:
I think we should run oldestdeps on PRs because if someone uses a function or python syntax which is too new then we should get warned about it.

As I see it, the builds we don't run on a PR should primarily be ones which are unlikely to be the fault of the change in the PR if they fail. I.e. devdeps because its likely to be upstream, conda because it's likely to be packaging etc.

Reply:
I'll change it.